src/components/grid/grid-item/grid-item.component.js

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable max-len */
+import React from 'react';
+import PropTypes from 'prop-types';
+import GridItemStyle from './grid-item.style';
+
+const GridItem = (props) => {
+  const {
+    children,
+    responsiveSettings,
+    gridColumnStart,
+    gridColumnEnd,
+    gridRowStart,
+    gridRowEnd
+  } = props;
+  const sortedByMaxWidth = responsiveSettings.sort(
+    (a, b) => ((a.maxWidth < b.maxWidth) ? 1 : -1)
+  );
+  return (
+    <GridItemStyle
+      { ...{
+        gridColumnStart,
+        gridColumnEnd,
+        gridRowStart,
+        gridRowEnd
+      } }
+      responsiveSettings={ sortedByMaxWidth }
+    >
+      {children}
+    </GridItemStyle>
+  );
+};
+
+GridItem.propTypes = {
+  /** Defines the Components(s) to be rendered within the GridItem */
+  children: PropTypes.node,
+  /** Defines the default starting column position of the GridItem within the GridContainer by referring to the grid line where the item begins */
```

Comment:
These comments are quite repetitive `Defines the default...` but there is no other way to provide a value. I think we can summarise these to be `starting column position of the...` , `ending column position...`. To compare [MUI](https://material-ui.com/api/button/). What do you think?